org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationPatching.cs

```diff
@@ -0,0 +1,442 @@
+            WriteToLog( "Is propResponse Null? : " + propResponse.IsNull().ToString() );
+            var props = new List<HubspotProperty>();
+            var tmbtProps = new List<HubspotProperty>();
+            var propsQry = JsonConvert.DeserializeObject<HSPropertyQueryResult>(propResponse.Content);
+            props = propsQry.results;
+            WriteToLog( "Is propsQry Null? : " + propsQry.IsNull().ToString() );
+            WriteToLog( "Is propsQry.results Null? : " + propsQry.results.IsNull().ToString() );
+            WriteToLog( "Is props Null? : " + props.IsNull().ToString() );
+            
+
+            //Filter to props in Rock Information Group (and Contact information group) 
+            props = props.Where(p => p.groupName == "rock_information" || p.groupName == "contactinformation").ToList();
+            //props = props.Where( p => p.groupName == "Contact information" ).ToList(); // - 6/26 sdlp - To update names, address, and phone
+            //tmbtProps = propsQry.results.Where(p => p.groupName == "rock_tmbt_information").ToList(); // ---------- Commented 6/25 sdlp
+            //Business Unit hs_all_assigned_business_unit_ids
+            //Save a list of the ones that are Rock attributes
+            //var attrs = props.Where(p => p.label.Contains("Rock Attribute ")).ToList(); // commented sdlp 7/15/24
+            RockContext _context = new RockContext();
+            //List<string> attrKeys = attrs.Select(hs => hs.label.Replace("Rock Attribute ", "")).ToList(); // commented sdlp 7/15/24
+            //var rockAttributes = new AttributeService(_context).Queryable().Where(a => a.EntityTypeId == 15 && attrKeys.Contains(a.Key)); //commented sdlp 7/15/24
+
+            Guid transactionTypeGuid = dataMap.GetString("ContributionTransactionType").AsGuid();
+            var transactionTypeDefinedValue = new DefinedValueService(_context).Get(transactionTypeGuid);
+            int transactionTypeValueId = transactionTypeDefinedValue.Id;
+
+            //Get List of all contacts from Hubspot
+            contacts = new List<HSContactResult>();
+            request_count = 0;
+
+            // TODO get all properties...
+            // email,firstname,lastname,gender,birthday,first_time_giving,first_visit_date,baptism,phone,rock_person_id,life_groups,enews_subscriber,first_time_serving,rock_account_created_date,lastmodifieddate
+            GetContacts( "https://api.hubapi.com/crm/v3/objects/contacts?limit=100&properties=rock_person_id" );
+            Debug.WriteLine("Contacts returned: " + contacts.Count());
+            WriteToLog(string.Format("Total Contacts to Match: {0}", contacts.Count()));
+            
+            PersonAliasService pa_svc = new PersonAliasService(_context);
+            FinancialTransactionService ft_svc = new FinancialTransactionService(_context);
+            AttributeValueService av_svc = new AttributeValueService(_context);
+            var dataViewService = new DataViewService( _context );
```

Comment:
Can these services be converted to using using statements so that they are disposed of properly if there is an error?